### ISSUE TITLE
URI environment variable needed.

### ISSUE THREAD

**Issue by almereyda** [ehdr/docker-known#1]
I just set this up behind an SSL offloader.

All links, those to static assetts and also POST action, are created for `http://`.

Unfortunately security restrictions will not allow to get static assetts from insecure sources.

Therefore a variable `URI=http(s)://known.example.com` would be very much appreciated.

**Comment by almereyda** [comment]
That goes together with:

```
docker logs known
docker-entrypoint.sh: Successfully connected to MySQL service
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 10.10.10.10. Set the 'ServerName' directive globally to suppress this message
```

**Comment by ehdr** [comment]
I can’t reproduce this at the moment, so more input would be appreciated. Is changing the Apache configuration enough to fix it, or would it also require changes in the Known PHP application (if so, you should also report an issue with the [upstream Known project](https://github.com/idno/idno))?

As for Apache configuration, adding an environment variable to set the `ServerName` at runtime would be fairly trivial. Are any other directives needed?

The Apache site configuration that is currently used for this image is intentionally as simple as possible, and you can have a look at it here:

https://github.com/ehdr/docker-known/blob/master/apache2/sites-available/known.conf

**Comment by almereyda** [comment]
You are right that the most easy way would be to render all site specific URIs as relative paths in upstream. Another solution would be to provide manually entering the site's FQDN and protocol. Which in return could then be used for Docker

---

Now that I take a look at the Apache Virtual Host configuration, if one could just rewrite the content that's serverd - through PHP - without even touching Known from the inside.

I see TLS as a mandatory feature for such an application.

**Comment by michielbdejong** [comment]
+1 I'm having the same problem running Known behind an SSL offloader, but using a different Dockerfile (which I [published](https://registry.hub.docker.com/u/indiehosters/known/) a few hours ago).

I think it must be an upstream problem, I'll comment here if/when I find a fix.

**Comment by michielbdejong** [comment]
Submitted upstream ticket ^

**Comment by almereyda** [comment]
yeah!

**Comment by michielbdejong** [comment]
As per the upstream ticket, this can be fixed by updating to 0.6.5.

**Comment by almereyda** [comment]
Merci beaucoup for taking care of it!
